nobl9/resource_budgetadjustment_test.go

```diff
@@ -3,29 +3,33 @@ package nobl9
 import (
 	"fmt"
 	"testing"
+	"time"
 
 	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
 	"github.com/nobl9/nobl9-go/manifest"
 )
 
 func TestAcc_Nobl9BudgetAdjustments(t *testing.T) {
+	futureTime := time.Now().AddDate(0, 0, 1)
 	cases := []struct {
-		name       string
-		configFunc func(string) string
+		name            string
+		configFunc      func(string, time.Time) string
+		firstEventStart time.Time
 	}{
-		{"single-event", testBudgetAdjustmentSingleEvent},
-		{"recurring-event", testBudgetAdjustmentRecurringEvent},
-		{"recurring-event-multiple-slos", testBudgetAdjustmentRecurringEventMultipleSlo},
+		{"single-event", testBudgetAdjustmentSingleEvent, futureTime},
+		{"recurring-event", testBudgetAdjustmentRecurringEvent, futureTime},
+		{"recurring-event-multiple-slos", testBudgetAdjustmentRecurringEventMultipleSlo, futureTime},
 	}
 
 	for _, tc := range cases {
 		t.Run(tc.name, func(t *testing.T) {
+			tc := tc
```

Comment:
Is this necessary?

Reply:
yes, because we are running the tests in parallel, range iterator works on pointers, and this can lead to unexpected results, because we can end up with different test case in our test

Reply:
Wasn't that fixed in GO 1.22? https://go.dev/blog/loopvar-preview

Reply:
well, it seems like it is 
```
For Go 1.22, we plan to change for loops to make these variables have per-iteration scope instead of per-loop scope. This change will fix the examples above, so that they are no longer buggy Go programs; it will end the production problems caused by such mistakes; and it will remove the need for imprecise tools that prompt users to make unnecessary changes to their code.
```
wasn't aware of that, thank you 😄

Reply:
removed in https://github.com/nobl9/terraform-provider-nobl9/pull/266/commits/063490cb6f9fc92730d80e703d1e282d963f1a43